tools/hilbert-station

```diff
@@ -64,66 +129,115 @@ trap 'exec 2>&4 1>&3 ' SIGHUP SIGINT SIGQUIT SIGPIPE SIGTERM EXIT RETURN
 exec 2>&1
 # 1>log.out
 
-function cmd_start_locking() {
+function cmd_start_locking() { ## @fn Start a lock to avoid parallel execution of this tool
     local pid
-    for pid in $(pidof -s "${TOOL}"); do
+    for pid in $(${PIDOF} -s "${TOOL}"); do
         if [[ $pid != $$ ]]; then
             ERROR "Process is already running with PID $pid"
-            exit 1
+            exit ${ERR_CODE_EXT_ERROR}
         fi
     done
 
     DEBUG "Starting exclusive (locked) usage of ${TOOL}..."
 
     # ON START:
     _prepare_locking
-    # Simplest example is avoiding running multiple instances of script: `exlock_now || exit 1`
+    # Simplest example is avoiding running multiple instances of script: `exlock_now || exit ${ERR_CODE_EXT_ERROR}`
     # Remember! Lock file is removed when one of the scripts exits and it is
     #           the only script holding the lock or lock is not acquired at all.
     local l
     if exlock_now
     then
-        l="$(ls -l "${LOCKFILE}" 2>&1)"
+        l="$(${LIST} -l "${LOCKFILE}" 2>&1)"
         DEBUG "Obtained exclusive lock of [$l]"
     else
-        l="$(ls -l "${LOCKFILE}" 2>&1)"
+        l="$(${LIST} -l "${LOCKFILE}" 2>&1)"
         DEBUG "Lockfile: [$l]"
         ERROR "Another user is already running ${TOOL}: please try again later!"
-        exit 2
+        exit ${ERR_CODE_EXT_ERROR}
+    fi
+    return $?
+}
+
+## The following is due to https://en.wikipedia.org/wiki/ANSI_escape_code
+#declare -rg ANSI_GREEN='\033[0;32m'
+declare -rg ANSI_GRAY='\033[0;37m'
```

Comment:
NOTE: try to detect support for colors in current terminal

Reply:
https://unix.stackexchange.com/questions/9957/how-to-check-if-bash-can-print-colors

Reply:
`tput colors`